sdk/eidos_sdk/system/agent_controller.py

```diff
@@ -286,7 +300,32 @@ async def _run_program(**_kwargs):
 
     async def get_process_info(self, process_id: str):
         latest_record = await self.get_latest_process_event(process_id)
-        return self.doc_to_response(latest_record)
+        if not latest_record:
+            return JSONResponse(dict(detail="Process not found"), 404)
+        elif (
+                latest_record.state == "unhandled_error"
+                or latest_record.state == "http_error"
+                or latest_record.state == "error"
+        ):
+            detail = latest_record.error_info
+            status_code = 500
+            if isinstance(latest_record.error_info, dict):
+                detail = latest_record.error_info.get("detail", latest_record.error_info)
+                status_code = latest_record.error_info.get("status_code", 500)
+            logger.info(f"Successfully retrieved stored error response, status_code={status_code}")
+            return JSONResponse(detail, status_code)
+        else:
+            return JSONResponse(
+                StateSummary(
+                    process_id=latest_record.record_id,
+                    state=latest_record.state,
+                    available_actions=self.get_available_actions(latest_record.state),
+                ).model_dump(),
+                200,
+            )
+
+    async def get_process_events(self, process_id: str):
```

Comment:
we will want some query args on this one eventually, but I don't think we need to bother with it now.